apps/avatax/src/modules/saleor/order.ts

```diff
@@ -47,3 +51,69 @@ export class SaleorOrder {
     return this.data.order.channel.taxConfiguration.pricesEnteredWithTax;
   }
 }
+
+interface ISaleorCancelledOrderEvent {
+  getChannelSlug(): string;
+  getAvataxId(): string;
+  getPrivateMetadata(): Array<{ key: string; value: string }>;
+}
+
+export class SaleorCancelledOrderEvent implements ISaleorCancelledOrderEvent {
+  private static schema = z.object({
+    order: z.object({
+      channel: z.object({
+        id: z.string(),
+        slug: z.string(),
+      }),
+      id: z.string(),
+      avataxId: z.string(),
+    }),
+    recipient: z.object({
+      privateMetadata: z.array(
+        z.object({
+          key: z.string(),
+          value: z.string(),
+        }),
+      ),
+    }),
+  });
+
+  private constructor(private data: z.infer<typeof SaleorCancelledOrderEvent.schema>) {}
+
+  getPrivateMetadata() {
+    return this.data.recipient.privateMetadata;
+  }
+
+  getChannelSlug() {
+    return this.data.order.channel.slug;
+  }
+
+  getAvataxId() {
+    return this.data.order.avataxId;
+  }
+
+  public static ParsingError = BaseError.subclass("AvataxAppSaleorOrderCancelledParsingError");
+
+  static create(payload: OrderCancelledPayloadFragment) {
+    if (!payload.order) {
+      return err(new OrderCancelPayloadOrderError("Insufficient order data"));
+    }
+
+    if (!payload.order.avataxId) {
+      return err(new OrderCancelNoAvataxIdError("No AvaTax id found in order"));
+    }
+
+    const parser = Result.fromThrowable(
+      SaleorCancelledOrderEvent.schema.parse,
+      SaleorCancelledOrderEvent.ParsingError.normalize,
+    );
+
+    const parsedPayload = parser(payload);
+
+    if (parsedPayload.isErr()) {
+      throw parsedPayload.error;
+    }
```

Comment:
why do you throw? The idea behind neverthrow is to avoid throwing. This method already returns errs and oks so here it can also return err